subprojects/instant-execution/src/main/kotlin/org/gradle/instantexecution/DefaultInstantExecution.kt

```diff
@@ -367,8 +374,12 @@ class DefaultInstantExecution internal constructor(
     }
 
     private
-    suspend fun DefaultReadContext.readGradleState(gradle: Gradle) {
+    suspend fun DefaultReadContext.readGradleState(gradle: GradleInternal) {
         withGradleIsolate(gradle) {
+            (gradle.settings.buildCache as BuildCacheConfigurationInternal).let { buildCache ->
```

Comment:
Instead of the cast, can we maybe add an `SettingsInternal.getBuildCache()` override?

Reply:
👍 done